File: add_two_numbers/add2numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        int r, x = 0, sum;
        ListNode *ln = new ListNode(0);
        ListNode *temp = ln;
        
        while(l1 != nullptr || l2 != nullptr) {
            if(l1 == nullptr) {
                sum = l2->val + x;
            }
            else if(l2 == nullptr) {
                sum = l1->val + x;
            }
            else {
                sum = l1->val + l2->val + x;
            }
            x = sum / 10;
            temp->next = new ListNode(sum % 10);
            if(l1 != nullptr) l1 = l1->next;
            if(l2 != nullptr) l2 = l2->next;
            temp = temp->next;
        }
        
        if(x > 0) temp->next = new ListNode(x);
        
        if(ln->next == nullptr)
            return ln;
        return ln->next;
    }
};

File: add_two_numbers/add2numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        result = ListNode()
        temp = result
        re = 0
        sum = 0
        
        while l1 != None or l2 != None or re:    
            if l1 == None and l2 == None:
                sum = re
            elif l1 == None:
                sum = re + l2.val
            elif l2 == None:
                sum = re + l1.val
            else:
                sum = l1.val + l2.val + re
            
            if l1 != None:
                l1 = l1.next 
            if l2 != None:
                l2 = l2.next
                
            val = sum % 10
            re = sum // 10
            
            temp.next = ListNode(val)
            temp = temp.next
        
        return result.next
